tests/test_components/test_medium.py

```diff
@@ -624,7 +623,7 @@ def test_nonlinear_medium(log_capture):
     # active materials
     with pytest.raises(ValidationError):
         med = td.Medium(
-            nonlinear_spec=td.NonlinearSpec(models=[td.TwoPhotonAbsorption(beta=-1 + 1j, n0=1)])
+            nonlinear_spec=td.NonlinearSpec(models=[td.TwoPhotonAbsorption(beta=-1, n0=1)])
```

Comment:
What exactly is happening with `beta`? It used to be possible to set it as complex, where the imaginary part means modulation of the *real* part of the permittivity? But TPA is purely absorption, so now instead we want beta to be purely real, while any dispersion would be coming from FCP?

Is the medium still unstable if the nonlinear effect is purely lossy like TPA?

Reply:
`beta` used to be complex, although typically a real beta would be used. I think you would use KerrNonlinearity instead if you wanted to modify the real part of the permittivity directly, or as you said you can have FCP dispersion coming from the same origin as the TPA.

Complex `beta` is implemented according to the docstring equation for TPA nonlinear polarization, which involves both the complex `n0` and `Re(n0)`. This equation is derived assuming real `beta`. For complex `beta`, we can use the same equation / implementation, but the physics may be less clear, and I cannot think of a use case. So, I deprecated complex `beta`.

TPA is lossy when `beta` is positive -- a negative value of `beta` would instead add energy in a nonlinear fashion, which may actually be used in some cases, although is an active medium and can lead to instability. The more general condition for stability involves the actual nonlinear polarization (and is what we use in the validator), and reduces to this when `n0` is real.